### ISSUE TITLE
init-hook no longer supports quoted code

### ISSUE THREAD

**Issue by mathiasritter** [PyCQA/pylint#7006]
# Bug Report

Since pylint version 2.14.2, the `init-hook` no longer supports quoted code. This is because with PR #6944, the function call to `utils._unquote` in `pylint/config/config_initialization.py` was removed: https://github.com/PyCQA/pylint/pull/6944/files#diff-dd2760aa48f0278b8c66e126913d22c0f9749cca892ef3c4104b957f4c899f68L48-R48

## Example

To reproduce this bug, I created a small [example repository](https://github.com/mathiasritter/pylint-init-hook) with the following contents:
```
src/
    mypackage/
        file1.py
        file2.py
.pylintrc
```

The contents of `file1.py` are:
```python
from mypackage.file2 import its_true
```

The contents of `file2.py` are:
```python
its_true = True
```

The contents of `.pylintrc` are:
```
[MASTER]
init-hook='import sys; sys.path.append("src")'  # Note that code is quoted
disable=missing-module-docstring,unused-import
```

The purpose of the `init-hook` is to add the `src` directory to the python path, so pylint can locate all imports correctly.

### Current Behaviour

When invoking pylint in the repository root with `src/mypackage/file1.py`, it is unable to import `mypackage.file2`:
```
❯ pylint src/mypackage/file1.py
************* Module file1
src/mypackage/file1.py:1:0: E0401: Unable to import 'mypackage.file2' (import-error)
```

However, pylint is able to import `mypackage.file2` if I either
- unquote the `init-hook` code or
- downgrade pylint to version <= 2.14.1

### Expected Behaviour

When running `pylint src/mypackage/file1.py`, pylint should be able to import `mypackage.file2`.

## Environment Information

pylint version 2.14.2
python version 3.10.5
macOS version 12.3.1

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue ! Thanks to your analysis we're suddenly figuring out the root cause of some issues that were opened for some time now 😉
